Build fails on macos due to missing endian.h

**Operating system**

macOS Monterey version 12.4

**Cmake version**

```
cmake --version
cmake version 3.23.2
```

**Ninja version**

```
ninja --version
1.11.0
```

**Compiler name and version**

```
clang --version
Apple clang version 13.1.6 (clang-1316.0.21.2.5)
Target: arm64-apple-darwin21.5.0
```

**Full cmake and/or ninja output**

```
$ cmake -G Ninja -DENABLE_CCACHE=0 -DCMAKE_BUILD_TYPE=RelWithDebInfo -S . -B build/
-- The C compiler identification is Clang 14.0.6
-- The CXX compiler identification is Clang 14.0.6
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /opt/homebrew/opt/llvm/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/homebrew/opt/llvm/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/homebrew/opt/llvm/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using compiler:
Homebrew clang version 14.0.6
Target: arm64-apple-darwin21.5.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
-- Using linker: <default>
-- Using archiver: /opt/homebrew/opt/llvm/bin/llvm-ar
-- Using ranlib: /opt/homebrew/opt/llvm/bin/llvm-ranlib
-- Using install-name-tool: /opt/homebrew/opt/llvm/bin/llvm-install-name-tool
-- Using objcopy: /opt/homebrew/opt/llvm/bin/llvm-objcopy
-- Using strip: /opt/homebrew/opt/llvm/bin/llvm-strip
-- Using ccache: no (disabled via configuration)
-- Found Git: /opt/homebrew/bin/git (found version "2.37.0")
-- HEAD's commit hash 5c8b9e4a147c89e29bbf23f3fb883ef7150913f6
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)
	modified:   contrib/icu (new commits, modified content, untracked content)

no changes added to commit (use "git add" and/or "git commit -a")
-- CMAKE_BUILD_TYPE: RelWithDebInfo
-- Performing Test HAS_RESERVED_IDENTIFIER
-- Performing Test HAS_RESERVED_IDENTIFIER - Success
-- Performing Test HAS_SUGGEST_DESTRUCTOR_OVERRIDE
-- Performing Test HAS_SUGGEST_DESTRUCTOR_OVERRIDE - Success
-- Performing Test HAS_SUGGEST_OVERRIDE
-- Performing Test HAS_SUGGEST_OVERRIDE - Success
-- Performing Test HAS_USE_CTOR_HOMING
-- Performing Test HAS_USE_CTOR_HOMING - Success
-- No official build: A checksum hash will not be added to the clickhouse executable
-- Default libraries: -nodefaultlibs  -lc -lm -lpthread -ldl
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Unit tests are enabled
-- Building for: Darwin-21.5.0 arm64  ;
    USE_STATIC_LIBRARIES=ON
    SPLIT_SHARED=
-- Adding contrib module miniselect (configuring with miniselect-cmake)
-- Adding contrib module pdqsort (configuring with pdqsort-cmake)
-- Adding contrib module sparsehash-c11 (configuring with sparsehash-c11-cmake)
-- Adding contrib module abseil-cpp (configuring with abseil-cpp-cmake)
-- Adding contrib module magic_enum (configuring with magic-enum-cmake)
-- Adding contrib module boost (configuring with boost-cmake)
-- Adding contrib module cctz (configuring with cctz-cmake)
-- Packaging with tzdata version: 2022a
-- Adding contrib module consistent-hashing (configuring with consistent-hashing)
-- Adding contrib module dragonbox (configuring with dragonbox-cmake)
-- Adding contrib module vectorscan (configuring with vectorscan-cmake)
-- Not using vectorscan
-- Adding contrib module jemalloc (configuring with jemalloc-cmake)
CMake Warning at contrib/jemalloc-cmake/CMakeLists.txt:20 (message):
  jemalloc support on non-linux is EXPERIMENTAL


-- jemalloc malloc_conf: oversize_threshold:0,muzzy_decay_ms:5000,dirty_decay_ms:5000
-- Adding contrib module libcpuid (configuring with libcpuid-cmake)
-- Not using cpuid
-- Adding contrib module libdivide (configuring with libdivide)
-- Adding contrib module libmetrohash (configuring with libmetrohash)
-- Adding contrib module lz4 (configuring with lz4-cmake)
-- Adding contrib module murmurhash (configuring with murmurhash)
-- Adding contrib module replxx (configuring with replxx-cmake)
-- Adding contrib module unixodbc (configuring with unixodbc-cmake)
-- ODBC is only supported on Linux
-- Not using ODBC
-- Adding contrib module nanodbc (configuring with nanodbc-cmake)
-- Adding contrib module capnproto (configuring with capnproto-cmake)
-- Adding contrib module yaml-cpp (configuring with yaml-cpp-cmake)
-- Adding contrib module re2 (configuring with re2-cmake)
-- Adding contrib module xz (configuring with xz-cmake)
-- Adding contrib module brotli (configuring with brotli-cmake)
-- Adding contrib module double-conversion (configuring with double-conversion-cmake)
-- Adding contrib module boringssl (configuring with boringssl-cmake)
-- Adding contrib module poco (configuring with poco-cmake)
-- Using Poco::Crypto
-- Not using Poco::Data::ODBC
-- Adding contrib module croaring (configuring with croaring-cmake)
-- Adding contrib module zstd (configuring with zstd-cmake)
-- ZSTD VERSION 1.5.0
-- Adding contrib module zlib-ng (configuring with zlib-ng-cmake)
-- Adding contrib module bzip2 (configuring with bzip2-cmake)
-- Adding contrib module minizip-ng (configuring with minizip-ng-cmake)
-- Adding contrib module snappy (configuring with snappy-cmake)
-- Adding contrib module rocksdb (configuring with rocksdb-cmake)
-- Performing Test HAS_ARMV8_CRC
-- Performing Test HAS_ARMV8_CRC - Success
--  HAS_ARMV8_CRC yes
-- Performing Test HAVE_FALLOCATE
-- Performing Test HAVE_FALLOCATE - Failed
-- Performing Test HAVE_SYNC_FILE_RANGE_WRITE
-- Performing Test HAVE_SYNC_FILE_RANGE_WRITE - Failed
-- Performing Test HAVE_PTHREAD_MUTEX_ADAPTIVE_NP
-- Performing Test HAVE_PTHREAD_MUTEX_ADAPTIVE_NP - Failed
-- Looking for malloc_usable_size
-- Looking for malloc_usable_size - not found
-- Adding contrib module thrift (configuring with thrift-cmake)
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for fcntl.h
-- Looking for fcntl.h - found
-- Looking for getopt.h
-- Looking for getopt.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for netdb.h
-- Looking for netdb.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for signal.h
-- Looking for signal.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for sys/ioctl.h
-- Looking for sys/ioctl.h - found
-- Looking for sys/param.h
-- Looking for sys/param.h - found
-- Looking for sys/resource.h
-- Looking for sys/resource.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/un.h
-- Looking for sys/un.h - found
-- Looking for poll.h
-- Looking for poll.h - found
-- Looking for sys/poll.h
-- Looking for sys/poll.h - found
-- Looking for sys/select.h
-- Looking for sys/select.h - found
-- Looking for sched.h
-- Looking for sched.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for strings.h
-- Looking for strings.h - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - not found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for sched_get_priority_max
-- Looking for sched_get_priority_max - found
-- Looking for sched_get_priority_min
-- Looking for sched_get_priority_min - found
-- Performing Test STRERROR_R_CHAR_P
-- Performing Test STRERROR_R_CHAR_P - Failed
-- Adding contrib module arrow (configuring with arrow-cmake)
-- Looking for strtof_l
-- Looking for strtof_l - not found
-- Looking for strtoull_l
-- Looking for strtoull_l - not found
-- CMAKE_CXX_FLAGS:  -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8-a+crc+simd+crypto+dotprod+ssbs -fasynchronous-unwind-tables -ffile-prefix-map=/tmp/ClickHouse=. -falign-functions=32 -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -w -std=c++11 -Wall -pedantic -Werror -Wextra -Wno-unused-parameter -stdlib=libc++
-- Proceeding with version: 1.12.0.372
-- Adding contrib module avro (configuring with avro-cmake)
-- Adding contrib module protobuf (configuring with protobuf-cmake)
-- Adding contrib module openldap (configuring with openldap-cmake)
-- Adding contrib module grpc (configuring with grpc-cmake)
CMake Warning (dev) at contrib/grpc/CMakeLists.txt:49 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'gRPC_BUILD_CSHARP_EXT'.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Looking for res_servicename
-- Looking for res_servicename - not found
-- Looking for res_servicename in resolv
-- Looking for res_servicename in resolv - found
-- Performing Test IOS
-- Performing Test IOS - Failed
-- Performing Test IOS_V10
-- Performing Test IOS_V10 - Failed
-- Performing Test MACOS_V1012
-- Performing Test MACOS_V1012 - Success
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for socket
-- Looking for socket - found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file arpa/nameser_compat.h
-- Looking for include file arpa/nameser_compat.h - found
-- Looking for include file arpa/nameser.h
-- Looking for include file arpa/nameser.h - found
-- Looking for include file assert.h
-- Looking for include file assert.h - found
-- Looking for include file errno.h
-- Looking for include file errno.h - found
-- Looking for include file limits.h
-- Looking for include file limits.h - found
-- Looking for include file malloc.h
-- Looking for include file malloc.h - not found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Looking for include file net/if.h
-- Looking for include file net/if.h - found
-- Looking for include file socket.h
-- Looking for include file socket.h - not found
-- Looking for include file stdbool.h
-- Looking for include file stdbool.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file stropts.h
-- Looking for include file stropts.h - not found
-- Looking for include file sys/uio.h
-- Looking for include file sys/uio.h - found
-- Looking for include file time.h
-- Looking for include file time.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include files sys/types.h, netinet/tcp.h
-- Looking for include files sys/types.h, netinet/tcp.h - found
-- Looking for include files winsock2.h, windows.h
-- Looking for include files winsock2.h, windows.h - not found
-- Looking for 3 include files winsock2.h, ..., windows.h
-- Looking for 3 include files winsock2.h, ..., windows.h - not found
-- Looking for include files winsock.h, windows.h
-- Looking for include files winsock.h, windows.h - not found
-- Looking for include file windows.h
-- Looking for include file windows.h - not found
-- Performing Test HAVE_SOCKLEN_T
-- Performing Test HAVE_SOCKLEN_T - Success
-- Performing Test HAVE_TYPE_SOCKET
-- Performing Test HAVE_TYPE_SOCKET - Failed
-- Performing Test HAVE_BOOL_T
-- Performing Test HAVE_BOOL_T - Success
-- Performing Test HAVE_SSIZE_T
-- Performing Test HAVE_SSIZE_T - Success
-- Performing Test HAVE_LONGLONG
-- Performing Test HAVE_LONGLONG - Success
-- Performing Test HAVE_SIG_ATOMIC_T
-- Performing Test HAVE_SIG_ATOMIC_T - Success
-- Performing Test HAVE_STRUCT_ADDRINFO
-- Performing Test HAVE_STRUCT_ADDRINFO - Success
-- Performing Test HAVE_STRUCT_IN6_ADDR
-- Performing Test HAVE_STRUCT_IN6_ADDR - Success
-- Performing Test HAVE_STRUCT_SOCKADDR_IN6
-- Performing Test HAVE_STRUCT_SOCKADDR_IN6 - Success
-- Performing Test HAVE_STRUCT_SOCKADDR_STORAGE
-- Performing Test HAVE_STRUCT_SOCKADDR_STORAGE - Success
-- Performing Test HAVE_STRUCT_TIMEVAL
-- Performing Test HAVE_STRUCT_TIMEVAL - Success
-- Looking for AF_INET6
-- Looking for AF_INET6 - found
-- Looking for O_NONBLOCK
-- Looking for O_NONBLOCK - found
-- Looking for FIONBIO
-- Looking for FIONBIO - found
-- Looking for SIOCGIFADDR
-- Looking for SIOCGIFADDR - found
-- Looking for MSG_NOSIGNAL
-- Looking for MSG_NOSIGNAL - found
-- Looking for PF_INET6
-- Looking for PF_INET6 - found
-- Looking for SO_NONBLOCK
-- Looking for SO_NONBLOCK - not found
-- Looking for CLOCK_MONOTONIC
-- Looking for CLOCK_MONOTONIC - found
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID - Success
-- Performing Test HAVE_LL
-- Performing Test HAVE_LL - Success
-- Looking for bitncmp
-- Looking for bitncmp - not found
-- Looking for closesocket
-- Looking for closesocket - not found
-- Looking for CloseSocket
-- Looking for CloseSocket - not found
-- Looking for connect
-- Looking for connect - found
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for freeaddrinfo
-- Looking for freeaddrinfo - found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getenv
-- Looking for getenv - found
-- Looking for gethostbyaddr
-- Looking for gethostbyaddr - found
-- Looking for gethostname
-- Looking for gethostname - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for getservbyport_r
-- Looking for getservbyport_r - not found
-- Looking for getservbyname_r
-- Looking for getservbyname_r - not found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for if_indextoname
-- Looking for if_indextoname - found
-- Looking for inet_net_pton
-- Looking for inet_net_pton - found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Looking for inet_pton
-- Looking for inet_pton - found
-- Looking for ioctl
-- Looking for ioctl - found
-- Looking for ioctlsocket
-- Looking for ioctlsocket - not found
-- Looking for IoctlSocket
-- Looking for IoctlSocket - not found
-- Looking for recv
-- Looking for recv - found
-- Looking for recvfrom
-- Looking for recvfrom - found
-- Looking for send
-- Looking for send - found
-- Looking for setsockopt
-- Looking for setsockopt - found
-- Looking for socket
-- Looking for socket - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for strcmpi
-- Looking for strcmpi - not found
-- Looking for strdup
-- Looking for strdup - found
-- Looking for stricmp
-- Looking for stricmp - not found
-- Looking for strncasecmp
-- Looking for strncasecmp - found
-- Looking for strncmpi
-- Looking for strncmpi - not found
-- Looking for strnicmp
-- Looking for strnicmp - not found
-- Looking for writev
-- Looking for writev - found
-- Looking for __system_property_get
-- Looking for __system_property_get - not found
-- Adding contrib module msgpack-c (configuring with msgpack-c-cmake)
-- Adding contrib module wyhash (configuring with wyhash-cmake)
-- Adding contrib module cityhash102 (configuring with cityhash102)
-- Adding contrib module libfarmhash (configuring with libfarmhash)
-- Adding contrib module icu (configuring with icu-cmake)
-- Not using icu
-- Adding contrib module h3 (configuring with h3-cmake)
-- Adding contrib module mariadb-connector-c (configuring with mariadb-connector-c-cmake)
-- Build without mysqlclient (support for MYSQL dictionary source will be disabled)
-- Adding contrib module googletest (configuring with googletest-cmake)
-- Adding contrib module llvm (configuring with llvm-cmake)
-- Not using LLVM
-- Adding contrib module libxml2 (configuring with libxml2-cmake)
-- Adding contrib module aws;aws-c-common;aws-c-event-stream;aws-checksums (configuring with aws-s3-cmake)
-- Adding contrib module base64 (configuring with base64-cmake)
-- Adding contrib module simdjson (configuring with simdjson-cmake)
-- Adding contrib module rapidjson (configuring with rapidjson-cmake)
-- Adding contrib module fastops (configuring with fastops-cmake)
-- Not using fastops
-- Adding contrib module libuv (configuring with libuv-cmake)
-- Adding contrib module AMQP-CPP (configuring with amqpcpp-cmake)
-- Adding contrib module cassandra (configuring with cassandra-cmake)
-- Adding contrib module fmtlib (configuring with fmtlib-cmake)
-- Adding contrib module krb5 (configuring with krb5-cmake)
-- Adding contrib module cyrus-sasl (configuring with cyrus-sasl-cmake)
-- Adding contrib module libgsasl (configuring with libgsasl-cmake)
-- Adding contrib module librdkafka (configuring with librdkafka-cmake)
-- librdkafka with SASL support
-- librdkafka with SSL support
-- Adding contrib module nats-io (configuring with nats-io-cmake)
-- Adding contrib module libhdfs3 (configuring with libhdfs3-cmake)
-- Not using hdfs
-- Adding contrib module hive-metastore (configuring with hive-metastore-cmake)
-- Not using hive
-- Adding contrib module cppkafka (configuring with cppkafka-cmake)
-- Adding contrib module libpqxx (configuring with libpqxx-cmake)
-- Adding contrib module libpq (configuring with libpq-cmake)
-- Adding contrib module NuRaft (configuring with nuraft-cmake)
-- Adding contrib module fast_float (configuring with fast_float-cmake)
-- Adding contrib module datasketches-cpp (configuring with datasketches-cpp-cmake)
-- Adding contrib module hashidsxx (configuring with hashidsxx-cmake)
-- Adding contrib module libstemmer_c (configuring with libstemmer-c-cmake)
-- Adding contrib module wordnet-blast (configuring with wordnet-blast-cmake)
-- Adding contrib module lemmagen-c (configuring with lemmagen-c-cmake)
-- Adding contrib module nlp-data (configuring with nlp-data-cmake)
-- Adding contrib module cld2 (configuring with cld2-cmake)
-- Adding contrib module sqlite-amalgamation (configuring with sqlite-cmake)
-- Adding contrib module s2geometry (configuring with s2geometry-cmake)
-- Adding contrib module base-x (configuring with base-x-cmake)
-- Performing Test SUPPORTS_CXXFLAG_frame_larger_than=65536
-- Performing Test SUPPORTS_CXXFLAG_frame_larger_than=65536 - Success
-- Performing Test SUPPORTS_CFLAG_frame_larger_than=65536
-- Performing Test SUPPORTS_CFLAG_frame_larger_than=65536 - Success
-- Performing Test SUPPORTS_CXXFLAG_everything
-- Performing Test SUPPORTS_CXXFLAG_everything - Success
-- Performing Test SUPPORTS_CFLAG_everything
-- Performing Test SUPPORTS_CFLAG_everything - Success
-- Performing Test SUPPORTS_CXXFLAG_pedantic
-- Performing Test SUPPORTS_CXXFLAG_pedantic - Success
-- Performing Test SUPPORTS_CFLAG_pedantic
-- Performing Test SUPPORTS_CFLAG_pedantic - Success
-- Performing Test SUPPORTS_CXXFLAG_no_vla_extension
-- Performing Test SUPPORTS_CXXFLAG_no_vla_extension - Success
-- Performing Test SUPPORTS_CFLAG_no_vla_extension
-- Performing Test SUPPORTS_CFLAG_no_vla_extension - Success
-- Performing Test SUPPORTS_CXXFLAG_no_zero_length_array
-- Performing Test SUPPORTS_CXXFLAG_no_zero_length_array - Success
-- Performing Test SUPPORTS_CFLAG_no_zero_length_array
-- Performing Test SUPPORTS_CFLAG_no_zero_length_array - Success
-- Performing Test SUPPORTS_CXXFLAG_no_c11_extensions
-- Performing Test SUPPORTS_CXXFLAG_no_c11_extensions - Success
-- Performing Test SUPPORTS_CFLAG_no_c11_extensions
-- Performing Test SUPPORTS_CFLAG_no_c11_extensions - Success
-- Performing Test SUPPORTS_CXXFLAG_no_unused_command_line_argument
-- Performing Test SUPPORTS_CXXFLAG_no_unused_command_line_argument - Success
-- Performing Test SUPPORTS_CFLAG_no_unused_command_line_argument
-- Performing Test SUPPORTS_CFLAG_no_unused_command_line_argument - Success
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat_pedantic
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat_pedantic - Success
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat_pedantic
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat_pedantic - Success
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat
-- Performing Test SUPPORTS_CXXFLAG_no_cxx98_compat - Success
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat
-- Performing Test SUPPORTS_CFLAG_no_cxx98_compat - Success
-- Performing Test SUPPORTS_CXXFLAG_no_c99_extensions
-- Performing Test SUPPORTS_CXXFLAG_no_c99_extensions - Success
-- Performing Test SUPPORTS_CFLAG_no_c99_extensions
-- Performing Test SUPPORTS_CFLAG_no_c99_extensions - Success
-- Performing Test SUPPORTS_CXXFLAG_no_conversion
-- Performing Test SUPPORTS_CXXFLAG_no_conversion - Success
-- Performing Test SUPPORTS_CFLAG_no_conversion
-- Performing Test SUPPORTS_CFLAG_no_conversion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_ctad_maybe_unsupported
-- Performing Test SUPPORTS_CXXFLAG_no_ctad_maybe_unsupported - Success
-- Performing Test SUPPORTS_CFLAG_no_ctad_maybe_unsupported
-- Performing Test SUPPORTS_CFLAG_no_ctad_maybe_unsupported - Success
-- Performing Test SUPPORTS_CXXFLAG_no_deprecated_dynamic_exception_spec
-- Performing Test SUPPORTS_CXXFLAG_no_deprecated_dynamic_exception_spec - Success
-- Performing Test SUPPORTS_CFLAG_no_deprecated_dynamic_exception_spec
-- Performing Test SUPPORTS_CFLAG_no_deprecated_dynamic_exception_spec - Success
-- Performing Test SUPPORTS_CXXFLAG_no_disabled_macro_expansion
-- Performing Test SUPPORTS_CXXFLAG_no_disabled_macro_expansion - Success
-- Performing Test SUPPORTS_CFLAG_no_disabled_macro_expansion
-- Performing Test SUPPORTS_CFLAG_no_disabled_macro_expansion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_documentation_unknown_command
-- Performing Test SUPPORTS_CXXFLAG_no_documentation_unknown_command - Success
-- Performing Test SUPPORTS_CFLAG_no_documentation_unknown_command
-- Performing Test SUPPORTS_CFLAG_no_documentation_unknown_command - Success
-- Performing Test SUPPORTS_CXXFLAG_no_double_promotion
-- Performing Test SUPPORTS_CXXFLAG_no_double_promotion - Success
-- Performing Test SUPPORTS_CFLAG_no_double_promotion
-- Performing Test SUPPORTS_CFLAG_no_double_promotion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_exit_time_destructors
-- Performing Test SUPPORTS_CXXFLAG_no_exit_time_destructors - Success
-- Performing Test SUPPORTS_CFLAG_no_exit_time_destructors
-- Performing Test SUPPORTS_CFLAG_no_exit_time_destructors - Success
-- Performing Test SUPPORTS_CXXFLAG_no_float_equal
-- Performing Test SUPPORTS_CXXFLAG_no_float_equal - Success
-- Performing Test SUPPORTS_CFLAG_no_float_equal
-- Performing Test SUPPORTS_CFLAG_no_float_equal - Success
-- Performing Test SUPPORTS_CXXFLAG_no_global_constructors
-- Performing Test SUPPORTS_CXXFLAG_no_global_constructors - Success
-- Performing Test SUPPORTS_CFLAG_no_global_constructors
-- Performing Test SUPPORTS_CFLAG_no_global_constructors - Success
-- Performing Test SUPPORTS_CXXFLAG_no_missing_prototypes
-- Performing Test SUPPORTS_CXXFLAG_no_missing_prototypes - Success
-- Performing Test SUPPORTS_CFLAG_no_missing_prototypes
-- Performing Test SUPPORTS_CFLAG_no_missing_prototypes - Success
-- Performing Test SUPPORTS_CXXFLAG_no_missing_variable_declarations
-- Performing Test SUPPORTS_CXXFLAG_no_missing_variable_declarations - Success
-- Performing Test SUPPORTS_CFLAG_no_missing_variable_declarations
-- Performing Test SUPPORTS_CFLAG_no_missing_variable_declarations - Success
-- Performing Test SUPPORTS_CXXFLAG_no_nested_anon_types
-- Performing Test SUPPORTS_CXXFLAG_no_nested_anon_types - Success
-- Performing Test SUPPORTS_CFLAG_no_nested_anon_types
-- Performing Test SUPPORTS_CFLAG_no_nested_anon_types - Success
-- Performing Test SUPPORTS_CXXFLAG_no_packed
-- Performing Test SUPPORTS_CXXFLAG_no_packed - Success
-- Performing Test SUPPORTS_CFLAG_no_packed
-- Performing Test SUPPORTS_CFLAG_no_packed - Success
-- Performing Test SUPPORTS_CXXFLAG_no_padded
-- Performing Test SUPPORTS_CXXFLAG_no_padded - Success
-- Performing Test SUPPORTS_CFLAG_no_padded
-- Performing Test SUPPORTS_CFLAG_no_padded - Success
-- Performing Test SUPPORTS_CXXFLAG_no_return_std_move_in_cxx11
-- Performing Test SUPPORTS_CXXFLAG_no_return_std_move_in_cxx11 - Failed
-- Performing Test SUPPORTS_CFLAG_no_return_std_move_in_cxx11
-- Performing Test SUPPORTS_CFLAG_no_return_std_move_in_cxx11 - Failed
-- Flag -Wno-return-std-move-in-c++11 is unsupported
-- Flag -Wno-return-std-move-in-c++11 is unsupported
-- Performing Test SUPPORTS_CXXFLAG_no_shift_sign_overflow
-- Performing Test SUPPORTS_CXXFLAG_no_shift_sign_overflow - Success
-- Performing Test SUPPORTS_CFLAG_no_shift_sign_overflow
-- Performing Test SUPPORTS_CFLAG_no_shift_sign_overflow - Success
-- Performing Test SUPPORTS_CXXFLAG_no_sign_conversion
-- Performing Test SUPPORTS_CXXFLAG_no_sign_conversion - Success
-- Performing Test SUPPORTS_CFLAG_no_sign_conversion
-- Performing Test SUPPORTS_CFLAG_no_sign_conversion - Success
-- Performing Test SUPPORTS_CXXFLAG_no_switch_enum
-- Performing Test SUPPORTS_CXXFLAG_no_switch_enum - Success
-- Performing Test SUPPORTS_CFLAG_no_switch_enum
-- Performing Test SUPPORTS_CFLAG_no_switch_enum - Success
-- Performing Test SUPPORTS_CXXFLAG_no_undefined_func_template
-- Performing Test SUPPORTS_CXXFLAG_no_undefined_func_template - Success
-- Performing Test SUPPORTS_CFLAG_no_undefined_func_template
-- Performing Test SUPPORTS_CFLAG_no_undefined_func_template - Success
-- Performing Test SUPPORTS_CXXFLAG_no_unused_template
-- Performing Test SUPPORTS_CXXFLAG_no_unused_template - Success
-- Performing Test SUPPORTS_CFLAG_no_unused_template
-- Performing Test SUPPORTS_CFLAG_no_unused_template - Success
-- Performing Test SUPPORTS_CXXFLAG_no_vla
-- Performing Test SUPPORTS_CXXFLAG_no_vla - Success
-- Performing Test SUPPORTS_CFLAG_no_vla
-- Performing Test SUPPORTS_CFLAG_no_vla - Success
-- Performing Test SUPPORTS_CXXFLAG_no_weak_template_vtables
-- Performing Test SUPPORTS_CXXFLAG_no_weak_template_vtables - Success
-- Performing Test SUPPORTS_CFLAG_no_weak_template_vtables
-- Performing Test SUPPORTS_CFLAG_no_weak_template_vtables - Success
-- Performing Test SUPPORTS_CXXFLAG_no_weak_vtables
-- Performing Test SUPPORTS_CXXFLAG_no_weak_vtables - Success
-- Performing Test SUPPORTS_CFLAG_no_weak_vtables
-- Performing Test SUPPORTS_CFLAG_no_weak_vtables - Success
-- Performing Test SUPPORTS_CXXFLAG_no_thread_safety_negative
-- Performing Test SUPPORTS_CXXFLAG_no_thread_safety_negative - Success
-- Performing Test SUPPORTS_CFLAG_no_thread_safety_negative
-- Performing Test SUPPORTS_CFLAG_no_thread_safety_negative - Success
-- compiler C   = /opt/homebrew/opt/llvm/bin/clang  -fdiagnostics-color=always -Xclang -fuse-ctor-homing  -gdwarf-aranges -pipe -march=armv8-a+crc+simd+crypto+dotprod+ssbs -fasynchronous-unwind-tables -ffile-prefix-map=/tmp/ClickHouse=. -falign-functions=32  -fdiagnostics-absolute-paths -fexperimental-new-pass-manager -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wno-c11-extensions -Wno-unused-command-line-argument -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-ctad-maybe-unsupported -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -O2 -g -DNDEBUG -O3 -g -gdwarf-4
-- compiler CXX = /opt/homebrew/opt/llvm/bin/clang++  -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8-a+crc+simd+crypto+dotprod+ssbs -fasynchronous-unwind-tables -ffile-prefix-map=/tmp/ClickHouse=. -falign-functions=32 -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wno-c11-extensions -Wno-unused-command-line-argument -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-ctad-maybe-unsupported -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -O2 -g -DNDEBUG -O3 -g -gdwarf-4
-- LINKER_FLAGS =  -rdynamic -Wl,-U,_inside_main
-- /tmp/ClickHouse/src: Have 8196 megabytes of memory.
        Limiting concurrent linkers jobs to 2 and compiler jobs to 3 (system has 10 logical cores)
-- Will build ClickHouse 22.7.1.1 revision 54464
-- StorageFileLog is only supported on Linux
-- ClickHouse modes:
-- Server mode: ON
-- Client mode: ON
-- Local mode: ON
-- Self-extracting executable: OFF
-- Benchmark mode: ON
-- Extract from config mode: ON
-- Compressor mode: ON
-- Copier mode: ON
-- Format mode: ON
-- Obfuscator mode: ON
-- ODBC bridge mode: OFF
-- Library bridge mode: ON
-- ClickHouse install: ON
-- ClickHouse git-import: ON
-- ClickHouse keeper mode: ON
-- ClickHouse keeper-converter mode: ON
-- Clickhouse disks mode: ON
-- bash_completion will be written to /usr/local/share/bash-completion/completions
-- Target check already exists
-- /tmp/ClickHouse/utils: Have 8196 megabytes of memory.
        Limiting concurrent linkers jobs to 2 and compiler jobs to OFF (system has 10 logical cores)
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/ClickHouse/build
```

```
$ cmake --build build
[0/2] Re-checking globbed directories...
[4243/7693] Generating include_private/lib/krb5/error_tables/asn1_err.c...de_private/lib/krb5/error_tables/asn1_err.h, include_private/asn1_err.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=asn1_err.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/asn1_err.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=asn1_err.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/asn1_err.et
[4244/7693] Generating include_private/lib/krb5/error_tables/kdb5_err.c...de_private/lib/krb5/error_tables/kdb5_err.h, include_private/kdb5_err.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=kdb5_err.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/kdb5_err.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=kdb5_err.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/kdb5_err.et
[4245/7693] Generating include_private/lib/krb5/error_tables/k5e1_err.c...de_private/lib/krb5/error_tables/k5e1_err.h, include_private/k5e1_err.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=k5e1_err.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/k5e1_err.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=k5e1_err.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/k5e1_err.et
[4249/7693] Generating include_private/lib/krb5/error_tables/krb524_err...rivate/lib/krb5/error_tables/krb524_err.h, include_private/krb524_err.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=krb524_err.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/krb524_err.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=krb524_err.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/krb524_err.et
[4251/7693] Generating include_private/lib/krb5/error_tables/krb5_err.c...de_private/lib/krb5/error_tables/krb5_err.h, include_private/krb5_err.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=krb5_err.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/krb5_err.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=krb5_err.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/krb5_err.et
[4252/7693] Generating include_private/lib/krb5/error_tables/kv5m_err.c...de_private/lib/krb5/error_tables/kv5m_err.h, include_private/kv5m_err.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=kv5m_err.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/kv5m_err.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=kv5m_err.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/krb5/error_tables/kv5m_err.et
[4260/7693] Generating include_private/lib/gssapi/generic/gssapi_err_ge...sapi/generic/gssapi_err_generic.h, include_private/gssapi_err_generic.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=gssapi_err_generic.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/gssapi/generic/gssapi_err_generic.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=gssapi_err_generic.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/gssapi/generic/gssapi_err_generic.et
[4262/7693] Generating include_private/lib/gssapi/krb5/gssapi_err_krb5....te/lib/gssapi/krb5/gssapi_err_krb5.h, include_private/gssapi_err_krb5.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=gssapi_err_krb5.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/gssapi/krb5/gssapi_err_krb5.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=gssapi_err_krb5.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/lib/gssapi/krb5/gssapi_err_krb5.et
[4263/7693] Generating include_private/util/profile/prof_err.c, include_private/util/profile/prof_err.h, include_private/prof_err.h
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_h.awk outfile=prof_err.h /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/util/profile/prof_err.et
+ /usr/bin/awk -f /tmp/ClickHouse/contrib/krb5/src/util/et/et_c.awk outfile=prof_err.c textdomain= localedir= /tmp/ClickHouse/build/contrib/krb5-cmake/include_private/util/profile/prof_err.et
[5233/7693] Building CXX object utils/self-extracting-executable/CMakeFiles/pre_compressor.dir/compressor.cpp.o
FAILED: utils/self-extracting-executable/CMakeFiles/pre_compressor.dir/compressor.cpp.o
/opt/homebrew/opt/llvm/bin/clang++ -DHAS_RESERVED_IDENTIFIER -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -I/tmp/ClickHouse/build/includes/configs -I/tmp/ClickHouse/contrib/zstd/lib -isystem /tmp/ClickHouse/contrib/libcxx/include -isystem /tmp/ClickHouse/contrib/libcxx/src -isystem /tmp/ClickHouse/contrib/libcxxabi/include -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -pipe -march=armv8-a+crc+simd+crypto+dotprod+ssbs -fasynchronous-unwind-tables -ffile-prefix-map=/tmp/ClickHouse=. -falign-functions=32 -stdlib=libc++ -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -Wall -Wextra -Wframe-larger-than=65536 -Weverything -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wno-c11-extensions -Wno-unused-command-line-argument -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-ctad-maybe-unsupported -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -Wno-thread-safety-negative -O2 -g -DNDEBUG -O3 -g -gdwarf-4  -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk -mmacosx-version-min=10.15   -D OS_DARWIN -Werror -nostdinc++ -std=gnu++20 -MD -MT utils/self-extracting-executable/CMakeFiles/pre_compressor.dir/compressor.cpp.o -MF utils/self-extracting-executable/CMakeFiles/pre_compressor.dir/compressor.cpp.o.d -o utils/self-extracting-executable/CMakeFiles/pre_compressor.dir/compressor.cpp.o -c /tmp/ClickHouse/utils/self-extracting-executable/compressor.cpp
/private/tmp/ClickHouse/utils/self-extracting-executable/compressor.cpp:12:10: fatal error: 'endian.h' file not found
#include <endian.h>
         ^~~~~~~~~~
1 error generated.
[5244/7693] Building CXX object utils/keeper-bench/CMakeFiles/keeper-bench.dir/main.cpp.o
ninja: build stopped: subcommand failed.
```


Built using these instructions: https://clickhouse.com/docs/en/development/build-osx/#build-clickhouse

It seems the change that caused this was introduced here: https://github.com/ClickHouse/ClickHouse/pull/39065

macos does not have `endian.h`. It does have `machine/endian.h`. Perhaps something similar to https://debugjournal.tumblr.com/post/99661010362/missing-byteswaph-and-endianh-on-mac-os-x or https://github.com/Yubico/libfido2/blob/3050185c732a2b661b6d12b579517de868e55714/openbsd-compat/openbsd-compat.h#L18-L26= is needed.

--- a/utils/self-extracting-executable/compressor.cpp
+++ b/utils/self-extracting-executable/compressor.cpp
@@ -9,7 +9,18 @@
 #include <cerrno>
 #include <memory>
 #include <iostream>
+#if defined __APPLE__
+
+// dependencies
+#include <machine/endian.h>
+#include <libkern/OSByteOrder.h>
+
+// define 64 bit macros
+#define htole64(x) OSSwapHostToLittleInt64(x)
+
+#else
 #include <endian.h>
+#endif
 
 #include "types.h"
 

--- a/utils/self-extracting-executable/compressor.cpp
+++ b/utils/self-extracting-executable/compressor.cpp
@@ -9,7 +9,7 @@
 #include <cerrno>
 #include <memory>
 #include <iostream>
-#if defined __APPLE__
+#if defined OS_DARWIN
 
 // dependencies
 #include <machine/endian.h>

--- a/utils/self-extracting-executable/decompressor.cpp
+++ b/utils/self-extracting-executable/decompressor.cpp
@@ -1,6 +1,6 @@
 #include <zstd.h>
 #include <sys/mman.h>
-#if defined __APPLE__
+#if defined OS_DARWIN
 #include <sys/mount.h>
 #else
 #include <sys/statfs.h>
@@ -12,7 +12,7 @@
 #include <cstdio>
 #include <cstring>
 #include <iostream>
-#if defined __APPLE__
+#if defined OS_DARWIN
 
 // dependencies
 #include <machine/endian.h>

--- a/utils/self-extracting-executable/decompressor.cpp
+++ b/utils/self-extracting-executable/decompressor.cpp
@@ -12,7 +12,18 @@
 #include <cstdio>
 #include <cstring>
 #include <iostream>
+#if defined __APPLE__
+
+// dependencies
+#include <machine/endian.h>
+#include <libkern/OSByteOrder.h>
+
+// define 64 bit macros
+#define le64toh(x) OSSwapLittleToHostInt64(x)
+
+#else
 #include <endian.h>
+#endif
 
 #include "types.h"
 

